graftm/pplacer.py

```diff
@@ -50,16 +50,23 @@ def jplace_split(self, jplace_file, alias_hash):
 
         # Load the placement file
         placement_file = json.load(open(jplace_file))
-
+        
         # Parse the placements based on unique identifies appended to the end
         # of each read
         for placement in placement_file['placements']: # for each placement
             hash = {} # create an empty hash
             for alias in alias_hash: # For each alias, append to the 'place' list each read that identifier
-                hash = {'p': placement['p'],
-                        'nm': [nm for nm in placement['nm'] if nm[0].split('_')[-1] == alias]}
-                alias_hash[alias]['place'].append(hash)
-
+                nm_list = []
+                p = placement['p']
+                for nm in placement['nm']:
+                    if nm[0].split('_')[-1] == alias:
+                        nm_list.append(nm)
+                if any(nm_list):
+                    hash = {'p': p,
+                            'nm': nm_list}
+                    alias_hash[alias]['place'].append(hash)
```

Comment:
this is probably fine but slightly dangerous, because you are changing the same object that is being iterated over. I'd make a separate variable and build upon that.